File: python/tripservice_mechanical/TripService_Original.py
from TripDOA import TripDAO
from UserSession import UserSession
from UserNotLoggedInException import UserNotLoggedInException

class TripServiceOriginal:
  def getTripsByUser(self, user):
    loggeduser = UserSession.getInstance().getLoggedUser()
    isFriend = False
    tripList = []
    if loggeduser:
      for friend in user.getFriends():
        if friend is user:
          isFriend = True
          break;
      if isFriend:
        tripList = TripDAO.findTripsByUser(user)
      return tripList
    else:
      raise UserNotLoggedInException()

File: kotlin/trip-service-kata/src/main/kotlin/org/craftedsw/tripservicekata/TripService_Original.kt
package org.craftedsw.tripservicekata

import org.craftedsw.tripservicekata.exception.UserNotLoggedInException
import org.craftedsw.tripservicekata.trip.Trip
import org.craftedsw.tripservicekata.trip.TripDAO
import org.craftedsw.tripservicekata.user.User
import org.craftedsw.tripservicekata.user.UserSession
import java.util.*

class TripService_Original {

    fun getTripsByUser(user: User): List<Trip> {
        var tripList: List<Trip> = ArrayList<Trip>()
        val loggedUser: User? = UserSession.instance.loggedUser
        var isFriend: Boolean = false
        if (loggedUser != null) {
            for (friend in user.friends) {
                if (friend == loggedUser) {
                    isFriend = true
                    break
                }
            }
            if (isFriend) {
                tripList = TripDAO.findTripsByUser(user)
            }
            return tripList
        } else {
            throw UserNotLoggedInException()
        }
    }

}
